nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,221 @@
+          echoHook = {
+            id = "echo";
+            execute-command = "echo";
+            response-message = "Webhook is reachable!";
+          };
+        };
+        description = mdDoc ''
+          The actual configuration of which hooks will be served.
+
+          It cannot be empty otherwise your service won't start.
+          Read more on the [project homepage]
+          [project homepage]: https://github.com/adnanh/webhook#configuration
+        '';
+        example = {
+          redeployWebhook = {
+            id = "redeploy-webhook";
+            execute-command = "/var/scripts/redeploy.sh";
+            working-directory = "/var/webhook";
+          };
+        };
+        type = types.attrsOf (types.submodule ({ config, ... }: {
+          options = {
+            id = mkOption {
+              type = types.str;
+              default = config._module.args.name;
+              description = mdDoc "specifies the ID of your hook. This value is used to create the HTTP endpoint (http://yourserver:port/hooks/your-hook-id)";
+            };
+            execute-command = mkOption {
+              type = types.str;
+              description = mdDoc "specifies the command that should be executed when the hook is triggered";
+            };
+            working-directory = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = mdDoc "specifies the working directory that will be used for the script when it's executed";
+            };
+            response-message = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = mdDoc "specifies the string that will be returned to the hook initiator";
+            };
+            response-headers = mkOption {
+              type = types.nullOr (types.listOf
+                (types.submodule {
+                  options = {
+                    name = mkOption {
+                      type = types.str;
+                      example = "X-Example-Header";
+                      description = mdDoc "Header name";
+                    };
+                    value = mkOption {
+                      type = types.str;
+                      example = "it works";
+                      description = mdDoc "Header value";
+                    };
+                  };
+                })
+              );
+              default = null;
+              description = mdDoc "specifies the list of headers that will be returned in HTTP response for the hook";
+            };
+          };
+        }));
+      };
+
+      successHttpResponseCode = mkOption {
+        type = types.nullOr types.ints.positive;
+        default = null;
+        description = mdDoc "This specifies the HTTP status code to be returned upon a successful hook call.";
+      };
+
+      # TODO: add other options from https://github.com/adnanh/webhook/blob/master/docs/Hook-Definition.md
```

Comment:
Why not adopting a `freeformType`-style options to let people define everything they want without you needing to type everything?

Reply:
The hooks are now freeform